### PR TITLE
Support finer grained activation retry processing of an integration activation.

### PR DESCRIPTION
**Pull request by chirino**

### CODE REVIEW

File: controllers/src/main/java/com/redhat/ipaas/controllers/integration/online/ActivateHandler.java [hunk 1]
```diff
@@ -83,29 +83,54 @@ public StatusUpdate execute(Integration integration) {
...
+                    String gitHubUser = getGitHubUser();
+                    log.info("{} : Looked up GitHub user {}", getLabel(integration), gitHubUser);
+
+                    Map<String, byte[]> projectFiles = createProjectFiles(gitHubUser, integration);
+                    log.info("{} : Created project files", getLabel(integration));
+
+                    gitCloneUrl = ensureGitHubSetup(integration, getWebHookUrl(deployment, secret), projectFiles);
+                    log.info("{} : Updated GitHub repo {}", getLabel(integration), gitCloneUrl);
+                    currentStep = 1;
+
```
Comment (duchuyctlk) [inline controllers/src/main/java/com/redhat/ipaas/controllers/integration/online/ActivateHandler.java]: Did you miss a `break` statement here? Or did you do it with intention?

Comment (chirino) [inline controllers/src/main/java/com/redhat/ipaas/controllers/integration/online/ActivateHandler.java]: it's with intension.  I want it continue onto the next steps. But skip some on a retry.

File: controllers/src/main/java/com/redhat/ipaas/controllers/integration/online/ActivateHandler.java [hunk 2]
```diff
@@ -83,29 +83,54 @@ public StatusUpdate execute(Integration integration) {
...
+        if( currentStatus != Integration.Status.Pending ) {
+            currentStep = 0;
         }
 
-        Integration.Status status = openShiftService.isScaled(deployment) ?
-            Integration.Status.Activated : Integration.Status.Pending;
-        log.info("{} : Setting status to {}", getLabel(integration), status);
+        try {
+            String gitCloneUrl = null;
+            switch (currentStep) {
```
Comment (chirino) [inline controllers/src/main/java/com/redhat/ipaas/controllers/integration/online/ActivateHandler.java]: I really don't need breaks.  Perhaps I can add one at the end. :)

**Comment by rhuss** [comment]
@chirino I'm on it and about to unify #345 and this PR. I'm just merging it into my branch, so please don't change anything on this PR till I'm done  ;)

File: controllers/src/main/java/com/redhat/ipaas/controllers/integration/online/ActivateHandler.java (hunk 1 above)
Comment (duchuyctlk) [inline controllers/src/main/java/com/redhat/ipaas/controllers/integration/online/ActivateHandler.java]: Ok i see

**Comment by rhuss** [comment]
Superseded by the updated #345 . Please reopen when I'm wrong.

**Pull request closed without merging**